featurebyte/api/api_object_util.py

```diff
@@ -128,3 +159,113 @@ def is_server_mode() -> bool:
     """
     sdk_execution_mode = os.environ.get("FEATUREBYTE_SDK_EXECUTION_MODE")
     return sdk_execution_mode == "SERVER"
+
+
+def map_object_id_to_name(
+    object_map: Dict[Optional[ObjectId], str], object_id: Union[ObjectId, List[ObjectId]]
+) -> Union[Optional[str], List[Optional[str]]]:
+    """
+    Map list of object ids object names
+
+    Parameters
+    ----------
+    object_map: Dict[Optional[ObjectId], str],
+        Dict that maps ObjectId to name
+    object_id: Union[ObjectId, List[ObjectId]]
+        List of object ids to map, or object id to map
+
+    Returns
+    -------
+    Union[Optional[str], List[Optional[str]]]
+    """
+    if isinstance(object_id, list):
+        return [object_map.get(_id) for _id in object_id]
+    return object_map.get(object_id)
+
+
+def map_dict_list_to_name(
+    object_map: Dict[Optional[ObjectId], str],
+    object_id_field: str,
+    object_dict: Union[Dict[str, ObjectId], List[Dict[str, ObjectId]]],
+) -> Union[Optional[str], List[Optional[str]]]:
+    """
+    Map list of object dict to object names
+
+    Parameters
+    ----------
+    object_map: Dict[Optional[ObjectId], str],
+        Dict that maps ObjectId to name
+    object_id_field: str
+        Name of field in object dict to get object id from
+    object_dict: Union[Dict[str, ObjectId], List[Dict[str, ObjectId]]]
+        List of dict to map
+
+    Returns
+    -------
+    Union[Optional[str], List[Optional[str]]]
+    """
+    if isinstance(object_dict, list):
+        return [
+            object_map.get(_obj_dict.get(object_id_field))
+            for _obj_dict in object_dict
+            if _obj_dict.get(object_id_field)
+        ]
+    return object_map.get(object_dict.get(object_id_field))
+
+
+def to_request_func(response_dict: dict[str, Any], page: int) -> bool:
```

Comment:
nit: We could make this function private as it is not used outside this module.

Reply:
will update in follow up :)